ls1intum/Artemis
# `Exam mode`: Improve unified course rounding for exams

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
Even after the changes of #4703, Sentry spawns (new) issues of missing course information for calculation the score rounding. [See ARTEMIS-KN6](https://sentry.ase.in.tum.de/organizations/artemis/issues/20507/events/e446bdfa5687488c9a0e2171951bcf9d/?project=2) as an example.

### Description
I used this PR to refactor that logic a bit. Instead of using the course management service in order to retrieve the course using an additional server call, the course object is now included in the initial request to the summary api. In order to maintain the linkage between the course and the exam exercises, the `convertStudentExamFromServer` creates a new exercise group for the retrieved exercises. ([See this method](https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/entities/exercise.model.ts#L206) why this is needed).

### Steps for Testing
Prerequisites:
- 1 Student
- 1 Exam with exercises of each type

1. Participate in the exam as a student
2. Assess the exam
3. Review the results as a student
4. Make sure that no more errors are present in the console
5. Make sure that every shown result follows the course-wide rounding accuracy.

### Review Progress

#### Code Review
#### Manual Tests
